keyvi/include/keyvi/index/internal/index_auto_config.h

```diff
@@ -0,0 +1,65 @@
+/* * keyvi - A key value store.
+ *
+ * Copyright 2018 Hendrik Muhs<hendrik.muhs@gmail.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * index_auto_config.h
+ *
+ *  Created on: Apr 11, 2018
+ *      Author: hendrik
+ */
+
+#ifndef KEYVI_INDEX_INTERNAL_INDEX_AUTO_CONFIG_H_
+#define KEYVI_INDEX_INTERNAL_INDEX_AUTO_CONFIG_H_
+
+#include <algorithm>
+#include <string>
+#include <thread>  // NOLINT
+#include <unordered_map>
+
+#include "index/constants.h"
+
+#include "util/configuration.h"
+#include "util/os_utils.h"
+
+// #define ENABLE_TRACING
+#include "dictionary/util/trace.h"
+
+namespace keyvi {
+namespace index {
+namespace internal {
+
+class IndexAutoConfig final {
+ public:
+  static size_t MaxSegments() {
+    // increase file descriptors and leave some extra room, to be improved later
+    size_t file_descriptors = keyvi::util::OsUtils::TryIncreaseFileDescriptors();
+    return file_descriptors - 100;
```

Comment:
Magic number 100: As I understand, you don't want to use all the file descriptors. Can you please explain why did you choose 100?

Reply:
I agree, thats extremely vague. The problem is basically that I want to leave some room between the total limit and the setting, so that the client can have some open descriptors as well, e.g. because you create the index from a reading a open file. The other reason: compiling requires 2 open descriptors, so I need some room for that as well.

I think there is no perfect answer. I will improve the documentation. On the other side: Incrementing the file descriptors already helps, on Linux it's upped from 1024 (easy to hit) to 1 million, hitting 1 million segments == file descriptors is quite unlikely.

Reply:
Thanks for the answer, it's clear now.